webapp/packages/plugin-sql-editor-new/src/SQLEditor/SQLCodeEditorPanel/useSQLCodeEditorPanel.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+
+import { useCallback } from 'react';
+
+import { useExecutor, useObservableRef } from '@cloudbeaver/core-blocks';
+import { throttle } from '@cloudbeaver/core-utils';
+import type { ISQLEditorData } from '@cloudbeaver/plugin-sql-editor';
+import type { Transaction } from '@codemirror/state';
+import type { ViewUpdate } from '@codemirror/view';
+
+import type { IEditor } from '../SQLCodeEditor/useSQLCodeEditor';
+
+interface State {
+  highlightActiveQuery: () => void;
+  onQueryChange: (query: string) => void;
+  onUpdate: (update: ViewUpdate) => void;
+}
+
+export function useSQLCodeEditorPanel(data: ISQLEditorData, editor: IEditor) {
+
+  const state: State = useObservableRef(() => ({
+    highlightActiveQuery() {
+      this.editor.highlightActiveQuery(true);
+
+      const segment = this.data.activeSegment;
+
+      if (segment) {
+        this.editor.highlightActiveQuery(segment.begin, segment.end);
+      }
+    },
+    onQueryChange(query: string) {
+      this.data.setQuery(query);
+    },
+    onUpdate(update: ViewUpdate) {
+      const transactions = update.transactions.filter(t => t.isUserEvent('select') || t.isUserEvent('input'));
```

Comment:
seems like `const selectionTransactions = update.transactions.filter(t => t.selection !== undefined);` will cover all cases that we need